superset-frontend/src/explore/controlUtils/standardizedFormData.ts

```diff
@@ -0,0 +1,182 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  ensureIsArray,
+  getChartControlPanelRegistry,
+  QueryFormData,
+} from '@superset-ui/core';
+import {
+  ControlStateMapping,
+  standardizedState,
+  StandardizedFormDataInterface,
+} from '@superset-ui/chart-controls';
+import { getControlsState } from 'src/explore/store';
+import { getFormDataFromControls } from './getFormDataFromControls';
+
+export const sharedControls: Record<keyof standardizedState, string[]> = {
+  metrics: ['metric', 'metrics', 'metric_2'],
+  columns: ['groupby', 'columns', 'groupbyColumns', 'groupbyRows'],
+};
+export const publicControls = [
+  // time section
+  'granularity_sqla',
+  'time_grain_sqla',
+  'time_range',
+  // filters
+  'adhoc_filters',
+  // subquery limit(series limit)
+  'limit',
+  // order by clause
+  'timeseries_limit_metric',
+  'series_limit_metric',
+  // desc or asc in order by clause
+  'order_desc',
+  // outer query limit
+  'row_limit',
+  // x asxs column
+  'x_axis',
+  // advanced analytics - rolling window
+  'rolling_type',
+  'rolling_periods',
+  'min_periods',
+  // advanced analytics - time comparison
+  'time_compare',
+  'comparison_type',
+  // advanced analytics - resample
+  'resample_rule',
+  'resample_method',
+];
+
+export class StandardizedFormData {
+  private sfd: StandardizedFormDataInterface;
+
+  constructor(sourceFormData: QueryFormData) {
+    /*
+     * Support form_data for smooth switching between different viz
+     * */
+    const standardizedState = {
+      metrics: [],
+      columns: [],
+    };
+    const formData = Object.freeze(sourceFormData);
+    const reversedMap = StandardizedFormData.getReversedMap();
+
+    Object.entries(formData).forEach(([key, value]) => {
+      if (reversedMap.has(key)) {
+        standardizedState[reversedMap.get(key)].push(...ensureIsArray(value));
+      }
+    });
+
+    const memorizedFormData = Array.isArray(
+      formData?.standardizedorm_data?.memorizedFormData,
```

Comment:
```suggestion
      formData?.standardized_form_data?.memorizedFormData,
```

Looks like there's a typo here

Reply:
Thanks! it a typo.

Reply:
I have added a unit test for the memorizedFormData.